R/standard.R

```diff
@@ -1,3 +1,37 @@
+#' Retrieve rendered mighty code component
+#' @description Retrieves and renders a mighty code component, supporting
+#' both built-in standards and custom components from local files.
+#'
+#' @details Processes different component types based on file extension or
+#' component name. For R files, extracts and renders custom functions. For
+#' Mustache templates, creates components from template files. For standard
+#' names, retrieves and renders built-in standard components.
+#'
+#' @param code_component Character string specifying either a standard component name
+#'   or path to a custom component file (R or Mustache template)
+#' @param params named `list` of input parameters. Passed along to `mighty_component$render()`
+#' @seealso [get_rendered_standard()], [mighty_component], [mighty_component_rendered]
+#' @returns An object of class `mighty_component_rendered` containing the
+#'   rendered code template
+#' @examples
+#' get_rendered_component("ady", list(variable = "ASTDY", date = "ASTDT"))
+#' @export
+get_rendered_component <- function(code_component, params) {
+  file_type <- tolower(tools::file_ext(code_component))
+
+  switch(
+    file_type,
+    "r" = get_rendered_custom(code_component),
```

Comment:
The R–file branch ignores `params`, so any passed parameters aren't applied when rendering custom R components. Consider forwarding `params` to `get_rendered_custom()` or to the render method.
```suggestion
    "r" = get_rendered_custom(code_component, params),
```

Reply:
no, parameters cannot be passed to get_rendered_custom